packages/server-common/src/handlers/webhooks/contentful.handler.ts

```diff
@@ -5,11 +5,22 @@ import { SendMessageCommand, SQSClient } from '@aws-sdk/client-sqs';
 import { Logger } from '../../utils';
 import { validateContentfulRequest } from '../../utils/validate-contentful-request';
 
+type ContentfulActions =
+  | 'publish'
+  | 'unpublish'
+  | 'create'
+  | 'save'
+  | 'autosave'
+  | 'archive'
+  | 'unarchive'
+  | 'delete'
+  | 'complete';
+
 const getActionFromRequest = (
   request: lambda.Request<ContentfulWebhookPayload>,
-): 'publish' | 'unpublish' => {
+): ContentfulActions => {
   const actions = request.headers['x-contentful-topic']?.split('.') ?? [];
-  return actions[actions.length - 1] as 'publish' | 'unpublish';
+  return actions[actions.length - 1] as ContentfulActions;
 };
 
 const getDetailTypeFromRequest = (
```

Comment:
because of the cast in the last line (and generally how its written) we could still associate any type of event in contentful here and I don't see actions other than publish and unpublish being rejected as unsupported which I think they should be. You might want to play around with this function implementation to see if there is a way to do it without casting so that you are forced to say something like if actions i not publish or unpublish then throw